agent/http_test.go

```diff
@@ -435,42 +435,51 @@ func TestParseWait_InvalidIndex(t *testing.T) {
 
 func TestParseConsistency(t *testing.T) {
 	t.Parallel()
+	a := NewTestAgent(t.Name(), nil)
+	defer a.Shutdown()
 	resp := httptest.NewRecorder()
-	var b structs.QueryOptions
-
-	req, _ := http.NewRequest("GET", "/v1/catalog/nodes?stale", nil)
-	if d := parseConsistency(resp, req, &b); d {
-		t.Fatalf("unexpected done")
-	}
-
-	if !b.AllowStale {
-		t.Fatalf("Bad: %v", b)
-	}
-	if b.RequireConsistent {
-		t.Fatalf("Bad: %v", b)
-	}
-
-	b = structs.QueryOptions{}
-	req, _ = http.NewRequest("GET", "/v1/catalog/nodes?consistent", nil)
-	if d := parseConsistency(resp, req, &b); d {
-		t.Fatalf("unexpected done")
-	}
 
-	if b.AllowStale {
-		t.Fatalf("Bad: %v", b)
-	}
-	if !b.RequireConsistent {
-		t.Fatalf("Bad: %v", b)
+	var tests = []struct {
```

Comment:
two comments:

1. formatting

```go
tests := []struct {
	url                   string
	allowStale            bool
	wantAllowStale        bool
	wantRequireConsistent bool
}{
	{"/v1/catalog/nodes?stale", false, true, false},
	{"/v1/catalog/nodes?stale", true, true, false},
	{"/v1/catalog/nodes?consistent", false, false, true},
	{"/v1/catalog/nodes?consistent", true, false, true},
	{"/v1/catalog/nodes", false, false, false},
	{"/v1/catalog/nodes", true, true, false},
}

for _, tt := range tests {
	name := fmt.Sprintf("url=%v, HTTP.AllowStale=%v", tt.url, tt.allowStale)
	t.Run(name, func(t *testing.T) {
		a.srv.agent.config.HTTPConfig.AllowStale = tt.allowStale

		var q structs.QueryOptions
		req, _ := http.NewRequest("GET", tt.url, nil)
		if d := a.srv.parseConsistency(resp, req, &q); d {
			t.Fatalf("Failed to parse consistency.")
		}
		if got, want := q.AllowStale, tt.wantAllowStale; got != want {
			t.Fatalf("got allowStale %v want %v", got, want)
		}
		if got, want := q.RequireConsistent, tt.wantRequireConsistent; got != want {
			t.Fatalf("got requireConsistent %v want %v", got, want)
		}
	})
}
```

2. I think the fact that we need to change the server config for this test to work shows a wrong design. `parseConsistency` should not depend on the server configuration. This should be a utility function which is independent of the server. Can you just pass the `allowStale` parameter from the config into the function?

```go
func parseConsistency(resp http.ResponseWriter, req *http.Request, allowStale bool, opts *structs.QueryOptions) {...}
```